edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -273,15 +272,22 @@ def _is_jwt_cookie_and_session_user_mismatch(self, request):
         - Also adds monitoring details for mismatches.
         - Should only be called for JWT cookies.
         """
-        # adds early monitoring for the JWT LMS user_id
-        jwt_lms_user_id = self._get_and_monitor_jwt_cookie_lms_user_id(request)
-
         is_forgiving_jwt_cookies_enabled = get_setting(ENABLE_FORGIVING_JWT_COOKIES)
         # This toggle provides a temporary safety valve for rollout.
         if not is_forgiving_jwt_cookies_enabled:
             return False
 
-        # If we set the request user in middleware for JWT auth, then we'd actually be checking JWT vs JWT user id.
+        jwt_username, jwt_lms_user_id = self._get_unsafe_jwt_cookie_username_and_lms_user_id(request)
```

Comment:
**Note to reviewer**: This was moved after `is_forgiving_jwt_cookies_enabled` check, because if for any reason there is any issue with this code, disabling `is_forgiving_jwt_cookies_enabled` would be the only way to fix quickly.

Please mark resolved once read.